### ISSUE TITLE
Implement new pseudo-integrated testing strategy

### ISSUE THREAD

**Issue by jaredsilver** [empirical-org/Empirical-Core#3394]
Currently, our test coverage is quite lacking. We have some front-end coverage with Jest, and we have some backend coverage with RSpec, but we don't really have a way for the front-end tests to tell us if the backend breaks something. (This has been problematic because hardcoding JSON blobs to be consumed by Jest tests makes the tests completely useless if what's being delivered from the endpoints has changed. Multiple regressions have failed to be caught because of this.)

We're not certain we want to run an extensive integration test suite (e.g. w/ selenium), especially with the craziness of running React on Rails. However, we can significantly increase our coverage by dynamically generating mock data from the backend tests for use in the front end tests.

Here's the strategy:

For all critical API endpoints, write backend tests as we usually would. However, instead of merely asserting against the returned data, we will also _save_ the returned data into a tmp directory. This temporary data will now effectively serve as stubs for the front end tests.

We won't actually have the front end tests hit the server; we'll merely save the output from the server to be consumed later by the front end tests.

This should be significantly faster than running a full integration suite, and it also has the added benefit of scoping out exactly what the front-end edge cases are that need to be covered based on the back-end edge cases that we write tests for. (A problem we've had in the past is that the front end doesn't handle edge cases as rigorously as does the back end.)

Note that this approach won't do us much good with respect to posting back to the server, but we can worry about that in the next iteration of our testing strategy.

**Comment by jaredsilver** [comment]
Additional implementation notes:

We can `require` a JSON file in Jest, like so: `const rawScores = require('./__data__/raw_scores');` (assuming that we have a file named `raw_scores.json` in the `__data__` directory)

So, perhaps we can just have a nice mock for request that looks something like this:

```
jest.mock('request', () => ({
    get: jest.fn().mockReturnValue(/* stub data goes here */)
}))
```

**Comment by jaredsilver** [comment]
This has been experimentally implemented in #3540 🎉
